source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -172,6 +174,87 @@ void FilterConfigPerRoute::merge(const FilterConfigPerRoute& other) {
   }
 }
 
+// Constructor used for merging configurations from different levels (vhost, route, etc.)
+FilterConfigPerRoute::FilterConfigPerRoute(const FilterConfigPerRoute& less_specific,
+                                           const FilterConfigPerRoute& more_specific)
+    : context_extensions_(less_specific.context_extensions_),
+      check_settings_(more_specific.check_settings_), disabled_(more_specific.disabled_),
+      // Use most specific service override. If more specific has either service, use only that.
+      // This prevents having both services when one should override the other.
+      grpc_service_(more_specific.grpc_service_.has_value()
+                        ? more_specific.grpc_service_
+                        : (more_specific.http_service_.has_value() ? absl::nullopt
+                                                                   : less_specific.grpc_service_)),
+      http_service_(more_specific.http_service_.has_value()
+                        ? more_specific.http_service_
+                        : (more_specific.grpc_service_.has_value() ? absl::nullopt
+                                                                   : less_specific.http_service_)) {
+  // Merge context extensions from more specific configuration, overriding less specific ones.
+  for (const auto& extension : more_specific.context_extensions_) {
+    context_extensions_[extension.first] = extension.second;
+  }
+}
```

Comment:
This actually still did the merging. For example, if we only configured context_extensions in the route and didn't configure any http_service/grpc_service override, then the virtual host's override will still be used. We actually merged the route entry configuration and vhost configuation in this scenario.

I will perfer to only use the most specific's entry's configuration always. If the most specifc configuration entry has no override, then the configuration at the filter config will be used.

But I think this is a point we can discuss. I think the merging between route entry cfg and vhost cfg make the configuration is hard to understand and to manage for complex filters. But you may could provide other perspectives.